pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3837,6 +3838,11 @@ public void endReplaceSegments(String tableNameWithType, String segmentLineageEn
         tableNameWithType, segmentLineageEntryId);
   }
 
+  // Can be overridden to provide custom logic
```

Comment:
Can you please update the comment with some examples of what kinds of operations might be needed here?

Reply:
This is replacing segment, not refreshing segment.
+1 on @vvivekiyer 's comment. Please document an example operations that might be needed here. Also, do you want to also include `segmentsFrom`?

Reply:
Sure, added it